frontend/src/metabase-lib/Question.ts

```diff
@@ -775,13 +775,12 @@ class Question {
     const tableColumns = this.setting("table.columns");
     if (
       tableColumns &&
-      addedColumnNames.length > 0 &&
-      removedColumnNames.length === 0
+      (addedColumnNames.length > 0 || removedColumnNames.length > 0)
     ) {
       return this.updateSettings({
         "table.columns": [
           ...tableColumns.filter(
-            column => !addedColumnNames.includes(column.name),
+            column => !removedColumnNames.includes(column.name),
```

Comment:
Should we maybe extend the original check? It ensured we wouldn't duplicate columns when merging `tableColumns` and `addedColumnNames.map(...)` result. I think we might want to keep that? I'm thinking of something like:

```jsx
tableColumns.filter(
  column =>
    !addedColumnNames.includes(column.name) &&
    !removedColumnNames.includes(column.name)
)
```

Reply:
@ranquild would appreciate if you could give it a look; maybe you still remember what was happening in #24503

Reply:
There is e2e repro https://github.com/metabase/metabase/pull/24503/files#diff-e3b32eb50834bfcb0d107c2b06f89b004d6e5c313ef210db40c12922a1188ddcR191

Reply:
@kulyk We could just extend the original check, though it really shouldn't be necessary. The reason the columns were still there was because we weren't removing them. But extending the check shouldn't cause any hard either

Reply:
I added it back in. It shouldn't hurt anything anyways.